File: ta4j-core/src/main/java/org/ta4j/core/backtest/BacktestBarBuilderFactory.java
package org.ta4j.core.backtest;

import org.ta4j.core.api.series.BarBuilderFactory;
import org.ta4j.core.api.series.BarSeries;

/**
 * @author Lukáš Kvídera
 */
class BacktestBarBuilderFactory implements BarBuilderFactory {


  private BacktestBarBuilder backtestBarBuilder;


  @Override
  public BacktestBarBuilder createBarBuilder(final BarSeries series) {
    if (this.backtestBarBuilder == null) {
      this.backtestBarBuilder = new BacktestBarBuilder(series);
    }

    return this.backtestBarBuilder;
  }
}

File: ta4j-core/src/main/java/org/ta4j/core/backtest/BacktestBarBuilderFactory.kt
package org.ta4j.core.backtest

import org.ta4j.core.api.series.BarBuilderFactory
import org.ta4j.core.api.series.BarSeries

/**
 * @author Lukáš Kvídera
 */
internal class BacktestBarBuilderFactory : BarBuilderFactory {
    private var backtestBarBuilder: BacktestBarBuilder? = null


    override fun createBarBuilder(series: BarSeries): BacktestBarBuilder {
        if (backtestBarBuilder == null) {
            backtestBarBuilder = BacktestBarBuilder(series)
        }

        return backtestBarBuilder!!
    }
}
